config/utils.go

```diff
@@ -118,7 +125,16 @@ func contextsDirFromConfigDir(configDir string) string {
 
 func listYamlNames(dirName string) ([]string, error) {
 	file, err := os.Open(dirName)
-	if err != nil {
+
+	if os.IsNotExist(err) {
+		//if we cannot open the directory, create id and open again
+		_ = os.MkdirAll(dirName, 0750)
```

Comment:
This permission set seems big. What about only allowing user to read and write? 
```suggestion
		_ = os.MkdirAll(dirName, 0600)
```

Reply:
seems to work ok, It can be kept at 600